src/regions.c

```diff
@@ -214,20 +214,8 @@ regions_evacuate_output(struct output *output)
 {
 	assert(output);
 	struct view *view;
-	struct region *region;
-
 	wl_list_for_each(view, &output->server->views, link) {
-		wl_list_for_each(region, &output->regions, link) {
-			if (view->tiled_region == region) {
-				if (!view->tiled_region_evacuate) {
-					view->tiled_region_evacuate =
-						xstrdup(region->name);
-				}
-				/* Prevent carrying around a dangling pointer */
-				view->tiled_region = NULL;
-				break;
-			}
-		}
+		view_evacuate_regions(view, &output->regions);
```

Comment:
This could actually have an early out, e.g. something like
```c
if (!view->tiled_region) {
  continue;
}
```
before calling `view_evacuate_regions()`

Nit: I think `evacuate_region()` (singular) would better describe the new function, possibly even `try_evacuate_region()`?

An alternative would be to add a `struct output` pointer to `struct region`, then we wouldn't have to loop over all regions and could just do something like
```c
wl_list_for_each(view, &output->server->views, link) {
	if (view->tiled_region && view->tiled_region->output == output) {
		view->tiled_region_evacuate = xstrdup(view->tiled_region->name);
		view->tiled_region = NULL;
	}
}
```

Reply:
Good ideas; I will mull over these and come back to this another day (getting late here).

Reply:
How about `view_evacuate_any_region_of(struct view *view, struct output *output)`?

And I will add the "early out" as suggested.

Reply:
Or alternatively leave the loop+conditional where it is, and then `view_evacuate_region(struct view *view)` unconditionally evacuates `view->tiled_region`.

Reply:
> How about `view_evacuate_any_region_of(struct view *view, struct output *output)`?

A view can only have a single region so the `any_region_of` part seems a bit misleading to me.

You don't like the  "add `struct output` to `struct region`" idea? I think it would improve the readability of that function immensely and by making the complete inner loop obsolete.

Reply:
> You don't like the "add struct output to struct region" idea?
I do like the idea, it's a nice optimization and has precedent in labwc (e.g. `struct view` pointing back to `struct server`).
I was just hesitating as I'm not too familiar with the region code yet. I'm happy to make the change if you can give it some good testing and make sure I haven't inadvertently broken something.

Reply:
Just pushed a new version that (I think) incorporates all the suggestions.